test/csuite/wt11126_compile_config/main.c

```diff
@@ -0,0 +1,329 @@
+/*-
```

Comment:
A few suggestions for safety testing:
* From the existing Python metadata generate a test that would set every config key, then check that the values that you get from the compiled version are the same.
* Check every key's default value that you get if it's unset.
* Check incorrect inputs for various types (wrong values in place of integers or booleans or subconfigs).
* What happens if the user passes a string formatted as a compiled config? Since there's no distinction of the config type, we handle the compiled variant the same way as clear text one which can be unsafe.

Reply:
Agree that the test should verify that the implementation is doing the right thing with error checking. I'm not sure we necessarily need to check every single key, as they are automatically generated, but there should be enough coverage to ensure that we are doing the right thing.